---
title: "CTA1 mRNA and protein induction (Figure 3 and 4)"
author: "JY"
date: "2022-10-25 (updated `r Sys.Date()`)"
output: 
  html_notebook:
    code_folding: hide
    toc: true
    toc_float: true
---

```{r}
library(tidyverse)
library(ggtext)
#library(mgcv)
library(broom)
library(cowplot)
```


# _CTA1_ mRNA induction [Fig. 3A]
Experiment was performed on 2020/03/25-27, see [here](https://docs.google.com/spreadsheets/d/1dClNE3oDJyQQlZwcf_NoHs56Wi1G938K44yyCrwCpCQ/edit?usp=sharing) for experimental methods. Briefly, _C. glabrata_ cells were grown to mid-log phase (OD~0.5), washed and treated with either SC media (mock), 0mM Pi (no Pi) or 10 mM H2O2 for 45 minutes. RNA was extracted and qRT-PCR was performed with _ACT1_ used as a reference gene and the mock treated sample as the reference sample. The following dataset was extracted from the above spreadsheet.
```{r}
rna <- read_tsv("input/CTA1-mRNA-qPCR/20200327-CTA1-mRNA-Ct.txt") %>% 
  mutate(Treatment = fct_recode(Treatment, 
                                H2O2 = "10mM H2O2", 
                                `-Pi` = "no Pi", 
                                Mock = "untreated"))
# check the control wells
rna %>% filter(Group %in% c("noRT", "NTC"))
# get the subset for "unknowns" vs "standards"
rnaT <- rna %>% filter(Group == "unknown")
rnaS <- rna %>% filter(Group == "standard")
```
> C10 and D12 are problematic

Check the standard curve
```{r}
p <- rnaS %>% 
  ggplot(aes(x = log10(Dilution), y = Cp)) + 
  geom_point(aes(color = Target)) +
  scale_color_brewer(palette = "Set1") +
  theme_cowplot()
p
```
```{r}
rnaS %>% 
  #select(Target, Dilution, Cp) %>% 
  mutate(x = log10(Dilution)) %>% 
  group_by(Target) %>% 
  nest() %>% 
  mutate(test = map(data, function(df) lm(Cp ~ x, data = df)),
         slope = map(test, tidy)) %>% 
  unnest(slope) %>% 
  filter(term == "x") %>% 
  select(Target, slope = estimate, std.error, p.value)
  
```

> primer set C behaves better than primer set A

Visualize the Ct values for the unknown samples
```{r}
p <- rnaT %>% 
  ggplot(aes(x = Treatment, y = Cp)) + 
  geom_point(aes(color = Name), shape = 16, size = 2.5, position = position_dodge(width = 0.4)) + 
  facet_wrap(~Target) +
  theme_bw(base_size = 16)
p
```

Calculate ∆∆Ct. Because the dilution amount is the same across all samples, we don't have to make any adjustments. `r all(rnaT$Dilution == 100)`
```{r}
rna.dCt <- rnaT %>% 
  # calculate mean Ct for the reference gene
  filter(Target == "ACT1") %>% 
  group_by(Name, Treatment) %>% 
  summarize(refG = mean(Cp), .groups = "drop") %>% 
  # append the reference gene Ct mean to the original data
  right_join(filter(rnaT, Target != "ACT1")) %>% 
  # subtract the reference gene Ct mean from each target
  mutate(dCt = Cp - refG) %>% 
  select(-refG)

rna.ddCt <- rna.dCt %>% 
  # calculate mean dCt for the reference sample
  filter(Treatment == "Mock") %>% 
  group_by(Name, Target) %>% 
  summarize(refT = mean(dCt), .groups = "drop") %>% 
  right_join(rna.dCt) %>% 
  mutate(ddCt = dCt - refT, FC = 2^-ddCt) %>% 
  select(-refT)
```

Visualize fold changes relative to the mock treated sample and normalized by ACT1. Using CTA1 primer set A. Overall pattern is highly consistent. Primer set C data contains one outlier that is most likely due to experimental mistakes -- one -Pi sample showed >400 fold induction 
```{r}
treat_color <- c(`H<sub>2</sub>O<sub>2</sub>` = "#E69138", `-Pi` = "#9900FF", Mock = "#666666")
p.rna <- filter(rna.ddCt, Target == "CTA1_A") %>% 
  mutate(Strain = fct_recode(Name, `wild type 1` = "yH001", `wild type 2` = "yH002"),
         Treatment = fct_recode(Treatment, `H<sub>2</sub>O<sub>2</sub>` = "H2O2")) %>% 
  ggplot(aes(x = Treatment, y = FC)) +
  geom_point(aes(shape = Strain, color = Treatment), size = 3.5, alpha = 0.9,
             position = position_jitterdodge(dodge.width = 0.6, jitter.width = 0.2)) +
  scale_shape_discrete(solid = TRUE) +
  scale_color_manual(values = treat_color, guide = "none") +
  stat_summary(aes(group = Strain), fun = "mean", geom = "crossbar", color = "darkred",
               position = position_dodge(0.6), width = 0.4) +
  scale_y_log10() + ylab("CTA1 mRNA over mock") +
  theme_cowplot(line_size = 0.7, font_size = 16) +
  theme(legend.text = element_text(size = rel(0.8)),
        legend.title = element_text(size = rel(0.9)),
        legend.position = "top",
        axis.title.x = element_blank(),
        axis.text.x = element_markdown(size = rel(1.1)),
        axis.text.y = element_text(size = rel(1.1)))
ggsave("output/20221213-CTA1-mRNA-qRT-PCR.png", width = 4, height = 3.5)
```

Statistical tests
```{r}
lm <- rna.ddCt %>% filter(Target == "CTA1_A") %>% 
  mutate(Treatment = fct_relevel(Treatment, "Mock")) %>% 
  rename(Strain = Name) %>% 
  lm(ddCt ~ Treatment + Strain, data = .)
summary(lm)
```

# Cta1-GFP protein levels
## Common plotting functions
**Update 2022-11-08** 
This list is used as a shared plotting set up.
```{r}
p.timecourse <- list(
  stat_summary(fun = "mean", geom = "point", size = 3),
  stat_summary(fun.data = "mean_cl_boot", geom = "errorbar", width = 3),
  stat_smooth(aes(size = Genotype == "WT"), method = "loess", formula = 'y~x', se = FALSE),
  scale_x_continuous(breaks = seq(0,240,30)),
  scale_size_manual(values = c(0.8, 1.5), guide = "none"),
  labs(y = "Cta1-GFP protein level (a.u.)"),
  theme_cowplot(line_size = 0.7, font_size = 14),
  theme(legend.position=c(0.05,0.75), 
        #legend.box.background = element_rect(color = "black"),
        #legend.box.margin = margin(3,3,3,3),
        legend.text = element_text(face = 3), 
        legend.title = element_text(),
        axis.title = element_text(size = rel(1)), 
        axis.title.x = element_blank(),
        axis.text.x = element_text(hjust=0.5), axis.text = element_text(size = rel(1)),
  )
)
```

## Cta1 induction under -Pi and H<sub>2</sub>O<sub>2</sub> [Fig. 3B]

### Prepare data
```{r}
pi.dat.long <- read_csv("input/wt-Pi-H2O2/Long_data_CTA1-GFP.csv", col_types = cols()) %>% 
  mutate(new_time = as.numeric(gsub("min","",Time))) %>% 
  arrange(new_time, Treatment)
```

### Bin's version
```{r}
treat_color <- c(`2mM H2O2` = "#E69138", `0mM Pi` = "#9900FF", Control = "#666666")

p.gfp <- pi.dat.long %>%
  filter(!is.na(median)) %>% 
  ggplot(aes(x = new_time, y = median, color = Treatment)) + p.timecourse[-3] +
  stat_smooth(method = "loess", formula = 'y~x', se = FALSE) +
  scale_x_continuous(breaks = seq(0, 240, by = 60)) +
  scale_color_manual(values= treat_color, labels = c(bquote(~H[2]~O[2]), "-Pi", "Mock")) +
  theme(legend.text = element_text(size = rel(0.85), face = 1),
        legend.position="top",
        legend.title = element_blank())
plot_grid(p.rna, NULL, p.gfp, nrow = 1, rel_widths = c(1, 0.1, 1), align = "hv")
#ggsave("output/20221117-Pi-H2O2-timecourse.png", width = 4, height = 3.5)
ggsave("output/20221214-CTA1-rna-gfp-align.png", width = 8, height = 3.5)
```

## TF deletion mutants compared with wild type
### Prepare data
Import data
```{r}
files <- dir("input/msn24_gz_merge", pattern = "merge-*") # get the file names of the merged
files <- file.path("input/msn24_gz_merge", files)         # this appends the directory to the file names
files.s7y1 <- dir("input/s7y1_gz_merge", pattern = "merge-*") 
files.s7y1 <- file.path("input/s7y1_gz_merge", files.s7y1)  
files.m4s7 <- dir("input/m4s7_gz_merge", pattern = "merge-*") 
files.m4s7 <- file.path("input/m4s7_gz_merge", files.m4s7)
files.rapa <- dir("input/Rapa_gz_merge", pattern = "merge-*")
files.rapa <- file.path("input/Rapa_gz_merge", files.rapa)
files.r15_B1 <- dir("input/rim15_B1_gz_merge",pattern = "merge-*")
files.r15_B1 <- file.path("input/rim15_B1_gz_merge", files.r15_B1)
files.r15_B2 <- dir("input/rim15_B2_gz_merge", pattern = "merge-*")
files.r15_B2 <- file.path("input/rim15_B2_gz_merge",files.r15_B2)


names(files) <- c("Rep1-0Pi-Ctrl","Rep2-0Pi-Rep1-H2O2", "Rep3-0Pi-Rep2-H2O2", "Rep3-H2O2-Ctrl","Rep4-H2O2-Ctrl")
names(files.s7y1) <- c("Rep1-0Pi-Ctrl","Rep2-0Pi-Ctrl","Rep1-H2O2-Rep3-0Pi", "Rep2-H2O2-Rep4-0Pi", "Rep3-H2O2-Ctrl")
names(files.m4s7) <- c("Rep1-0Pi-Ctrl","Rep1-H2O2-Rep2-0Pi", "Rep2-H2O2-Rep3-0Pi", "Rep3-H2O2-Ctrl")
names(files.rapa) <- c("Rep1-Rapa-Ctrl","Rep2-Rapa-Ctrl","Rep3-Rapa-Ctrl")
names(files.r15_B1) <- c("Rep1-0Pi-Ctrl","Rep2-0Pi-Ctrl","Rep3-0Pi-Ctrl")
names(files.r15_B2) <- c("Rep1-0Pi-Ctrl","Rep2-0Pi-Ctrl","Rep3-0Pi-Ctrl")

tp <- c("0min",  "30min", "60min", "90min", "120min", "150min",  "180min", "210min", "240min") # ordered time points

# import and process the data to the format we want
dat.ord <- map_dfr(files, ~read_csv(., na = c("", "NA", "N/A"), col_types = cols()), .id = "Replicate") %>% 
  #bind_rows(.id = "Replicate") %>% 
  filter(`X Parameter` == "BL1-H", is.na(`Y Parameter`)) %>% 
  select(Replicate, Plate, Group, Sample, Count, median = `X Median`, peak = `X Peak`,
         sd = `X SD`, cv = `X %CV`, rCV = `X %rCV`) %>% 
  mutate(Group = ordered(Group, levels = tp)) %>% 
  arrange(Group)


# import and process the data to the format we want
dat.ord.s7y1 <- files.s7y1 %>% 
  map(~read_csv(., na = c("", "NA", "N/A"), col_types = cols())) %>% 
  bind_rows(.id = "Replicate") %>% 
  filter(`X Parameter` == "BL1-H", is.na(`Y Parameter`)) %>% 
  select(Replicate, Plate, Group, Sample, Count, median = `X Median`, peak = `X Peak`,
         sd = `X SD`, cv = `X %CV`, rCV = `X %rCV`) %>% 
  mutate(Group = ordered(Group, levels = tp)) %>% 
  arrange(Group)


# import and process the data to the format we want
dat.ord.m4s7 <- files.m4s7 %>% 
  map(~read_csv(., na = c("", "NA", "N/A"), col_types = cols())) %>% 
  bind_rows(.id = "Replicate") %>% 
  filter(`X Parameter` == "BL1-H", is.na(`Y Parameter`)) %>% 
  select(Replicate, Plate, Group, Sample, Count, median = `X Median`, peak = `X Peak`,
         sd = `X SD`, cv = `X %CV`, rCV = `X %rCV`) %>% 
  mutate(Group = ordered(Group, levels = tp)) %>% 
  arrange(Group)


# import and process the data to the format we want
dat.ord.rapa <- files.rapa %>% 
  map(~read_csv(., na = c("", "NA", "N/A"), col_types = cols())) %>% 
  bind_rows(.id = "Replicate") %>% 
  filter(`X Parameter` == "BL1-H", is.na(`Y Parameter`)) %>% 
  select(Replicate, Plate, Group, Sample, Count, median = `X Median`, peak = `X Peak`,
         sd = `X SD`, cv = `X %CV`, rCV = `X %rCV`) %>% 
  mutate(Group = ordered(Group, levels = tp)) %>% 
  arrange(Group)

# import and process the data to the format we want
dat.ord.r15_B1 <- files.r15_B1 %>% 
  map(~read_csv(., na = c("", "NA", "N/A"), col_types = cols())) %>% 
  bind_rows(.id = "Replicate") %>% 
  filter(`X Parameter` == "BL1-H", is.na(`Y Parameter`)) %>% 
  select(Replicate, Plate, Group, Sample, Count, median = `X Median`, peak = `X Peak`,
         sd = `X SD`, cv = `X %CV`, rCV = `X %rCV`) %>% 
  mutate(Group = ordered(Group, levels = tp)) %>% 
  arrange(Group)

# import and process the data to the format we want
dat.ord.r15_B2 <- files.r15_B2 %>% 
  map(~read_csv(., na = c("", "NA", "N/A"), col_types = cols())) %>% 
  bind_rows(.id = "Replicate") %>% 
  filter(`X Parameter` == "BL1-H", is.na(`Y Parameter`)) %>% 
  select(Replicate, Plate, Group, Sample, Count, median = `X Median`, peak = `X Peak`,
         sd = `X SD`, cv = `X %CV`, rCV = `X %rCV`) %>% 
  mutate(Group = ordered(Group, levels = tp)) %>% 
  arrange(Group)

```

Subset the dataset and merge with the genotype information
```{r}
# get genotype info
tr_files <- dir("input/msn24_treatment", pattern = "d0*")
tr_files <- file.path("input/msn24_treatment", tr_files)
names(tr_files) <- c("Rep1-0Pi-Ctrl","Rep2-0Pi-Rep1-H2O2", "Rep3-0Pi-Rep2-H2O2", "Rep3-H2O2-Ctrl","Rep4-H2O2-Ctrl")

tr_files.s7y1 <- dir("input/s7y1_treatment", pattern = "d0*")
tr_files.s7y1 <- file.path("input/s7y1_treatment", tr_files.s7y1)
names(tr_files.s7y1) <- c("Rep1-0Pi-Ctrl","Rep2-0Pi-Ctrl","Rep1-H2O2-Rep3-0Pi", "Rep2-H2O2-Rep4-0Pi", "Rep3-H2O2-Ctrl")

tr_files.m4s7 <- dir("input/m4s7_treatment", pattern = "d0*")
tr_files.m4s7 <- file.path("input/m4s7_treatment", tr_files.m4s7)
names(tr_files.m4s7) <- c("Rep1-0Pi-Ctrl","Rep1-H2O2-Rep2-0Pi", "Rep2-H2O2-Rep3-0Pi", "Rep3-H2O2-Ctrl")

tr_files.rapa <- dir("input/Rapa_treatment", pattern = "d0*")
tr_files.rapa <- file.path("input/Rapa_treatment", tr_files.rapa)
names(tr_files.rapa) <- c("Rep1-Rapa-Ctrl","Rep2-Rapa-Ctrl","Rep3-Rapa-Ctrl")

tr_files.r15_B1 <- dir("input/rim15_B1_treatment", pattern = "d0*")
tr_files.r15_B1 <- file.path("input/rim15_B1_treatment", tr_files.r15_B1)
names(tr_files.r15_B1) <- c("Rep1-0Pi-Ctrl","Rep2-0Pi-Ctrl","Rep3-0Pi-Ctrl")

tr_files.r15_B2 <- dir("input/rim15_B2_treatment", pattern = "d0*")
tr_files.r15_B2 <- file.path("input/rim15_B2_treatment", tr_files.r15_B2)
names(tr_files.r15_B2) <- c("Rep1-0Pi-Ctrl","Rep2-0Pi-Ctrl","Rep3-0Pi-Ctrl")


treat.fil <- tr_files %>% 
  map(~read_csv(., na = c("","NA","N/A"), col_types = cols())) %>% 
  bind_rows(.id = "Replicate") %>% 
  filter(!is.na(`group_name`)) %>% 
  select(Replicate, genotype = `group_name`, Sample, treat)


treat.fil.s7y1 <- tr_files.s7y1 %>% 
  map(~read_csv(., na = c("","NA","N/A"), col_types = cols())) %>% 
  bind_rows(.id = "Replicate") %>% 
  filter(!is.na(`group_name`)) %>% 
  select(Replicate, genotype = `group_name`, Sample, treat)


treat.fil.m4s7 <- tr_files.m4s7 %>% 
  map(~read_csv(., na = c("","NA","N/A"), col_types = cols())) %>% 
  bind_rows(.id = "Replicate") %>% 
  filter(!is.na(`group_name`)) %>% 
  select(Replicate, genotype = `group_name`, Sample, treat)

treat.fil.rapa <- tr_files.rapa %>% 
  map(~read_csv(., na = c("","NA","N/A"), col_types = cols())) %>% 
  bind_rows(.id = "Replicate") %>% 
  filter(!is.na(`group_name`)) %>% 
  select(Replicate, genotype = `group_name`, Sample, treat)

treat.fil.r15_B1 <- tr_files.r15_B1 %>% 
  map(~read_csv(., na = c("","NA","N/A"), col_types = cols())) %>% 
  bind_rows(.id = "Replicate") %>% 
  filter(!is.na(`group_name`)) %>% 
  select(Replicate, genotype = `group_name`, Sample, treat)

treat.fil.r15_B2 <- tr_files.r15_B2 %>% 
  map(~read_csv(., na = c("","NA","N/A"), col_types = cols())) %>% 
  bind_rows(.id = "Replicate") %>% 
  filter(!is.na(`group_name`)) %>% 
  select(Replicate, genotype = `group_name`, Sample, treat)


# combine genotype and treat variables into the flow data
data <- left_join(dat.ord, treat.fil, by = c("Replicate", "Sample"))
data.s7y1 <- left_join(dat.ord.s7y1, treat.fil.s7y1, by = c("Replicate", "Sample"))
data.m4s7 <- left_join(dat.ord.m4s7, treat.fil.m4s7, by = c("Replicate", "Sample"))
data.rapa <- left_join(dat.ord.rapa, treat.fil.rapa, by = c("Replicate","Sample"))
data.r15_B1 <- left_join(dat.ord.r15_B1, treat.fil.r15_B1, by = c("Replicate","Sample"))
data.r15_B2 <- left_join(dat.ord.r15_B2, treat.fil.r15_B2, by = c("Replicate","Sample"))

data.tmp <- add_row(data,data.s7y1)
data.cb <- add_row(data.tmp, data.m4s7)

data.r15 <- add_row(data.r15_B1,data.r15_B2)

```

### CTA1 induction during Nitrogen starvation & Rapa treatment

### CTA1 induction during -Pi [Fig. 4A]
Get the subset of no Pi data
```{r}
pi.dat <- data.cb %>% 
  filter(treat == "0Pi" & !genotype %in% c("skn7yap1_ko","msn2msn4_ko")) %>%  
  select(Time = Group, Sample, Plate, replicate = Replicate, genotype, median) %>%
  mutate(Genotype = fct_recode(genotype, WT = "wt", `msn2-` = "msn2_ko", `msn4-` = "msn4_ko", `yap1-` = "yap1_ko",
                               `skn7-` = "skn7_ko", `msn4- skn7-` = "msn4skn7_ko")) %>% 
  mutate(new_median =median/1000) %>% 
  mutate(new_time = as.numeric(gsub("min","",Time)))%>% 
  arrange(new_time, genotype, replicate)
```

*need to check the skn7 and yap1 data, I got them switched somehow*

#### Full time course

```{r}
genotype_color <- RColorBrewer::brewer.pal(8, "Dark2")[c(8, 7, 2, 3, 5, 1)] 
names(genotype_color) <- c("WT", "msn2-", "yap1-", "msn4-","skn7-", "msn4- skn7-")

pi.dat %>%
  filter(!is.na(new_median)) %>% 
  ggplot(aes(x = new_time, y = new_median, color = Genotype)) + p.timecourse +
  scale_color_manual(values= genotype_color, limits = names(genotype_color))
ggsave("output/20221108-trans-deletion-timecourse-noPi.png", width = 4.5, height = 4)
```
```{r}
genotype_color <- RColorBrewer::brewer.pal(8, "Dark2")[c(8, 7, 2, 3, 5, 1)] 
names(genotype_color) <- c("WT", "rim15-", "S986E", "S1038E","S986E S1038E", "S986E S1032E S1050E")

pi.dat <- data.r15 %>% 
  filter(treat == "0Pi" & genotype %in% c("wt", "S986E","S1038E","S986E_S1038E","S986E_S1032E_S1050E","rim15_ko")) %>%  
  select(Time = Group, Sample, Plate, replicate = Replicate, genotype, median) %>%
  mutate(Genotype = fct_recode(genotype, WT = "wt", `rim15-` = "rim15_ko", `S986E` = "S986E", `S1038E` = "S1038E",
                               `S986E S1038E` = "S986E_S1038E", `S986E S1032E S1050E` = "S986E_S1032E_S1050E")) %>% 
  mutate(new_median =median/1000) %>% 
  mutate(new_time = as.numeric(gsub("min","",Time)))%>% 
  arrange(new_time, genotype, replicate)

pi.dat %>%
  filter(!is.na(new_median)) %>% 
  ggplot(aes(x = new_time, y = new_median, color = Genotype)) + p.timecourse +
  scale_color_manual(values= genotype_color, limits = names(genotype_color))
ggsave("output/20230130-rim15-phosphomimetic-mutants-timecourse-noPi.png", width = 4.5, height = 4)
```

```{r}
treat_color <- RColorBrewer::brewer.pal(8, "Dark2")[c(8, 7, 2, 3, 5, 1, 4, 6)] 
names(treat_color) <- c("0Pi", "0Nitrogen", "50_Rapa", "62.5_Rapa","100_Rapa", "125_Rapa","150_Rapa")

pi.dat <- data.rapa %>% 
  filter(treat != "Ctrl") %>%  
  select(Time = Group, Sample, Plate, replicate = Replicate, genotype, median, treat) %>% 
  mutate(new_median =median/1000) %>% 
  mutate(new_time = as.numeric(gsub("min","",Time)))%>% 
  arrange(new_time, treat, replicate)

new.pi.course <- list(
  stat_summary(fun = "mean", geom = "point", size = 3),
  stat_summary(fun.data = "mean_cl_boot", geom = "errorbar", width = 3),
  stat_smooth(aes(size = genotype == "wt"), method = "loess", formula = 'y~x', se = FALSE),
  scale_x_continuous(breaks = seq(0,240,30)),
  scale_size_manual(values = c(0.8, 1.5), guide = "none"),
  labs(y = "Cta1-GFP protein level (a.u.)"),
  theme_cowplot(line_size = 0.7, font_size = 14),
  theme(legend.position=c(0.05,0.75), 
        #legend.box.background = element_rect(color = "black"),
        #legend.box.margin = margin(3,3,3,3),
        legend.text = element_text(face = 3), 
        legend.title = element_text(),
        axis.title = element_text(size = rel(1)), 
        axis.title.x = element_blank(),
        axis.text.x = element_text(hjust=0.5), axis.text = element_text(size = rel(1)),
  )
)

pi.dat %>%
  filter(!is.na(new_median)) %>% 
  ggplot(aes(x = new_time, y = new_median, color = treat)) + new.pi.course 
ggsave("output/20230130-rapa-0nitrogen-timecourse-noPi.png", width = 4.5, height = 4)
```


#### Basal vs relative induction levels [Fig. S?]
From the graph above, we can see that the tf∆ mutants affected both the basal expression level (at 0 min) and the fold change (relative to the 0'). To further dissect the effects on these two components, we will make some additional plots below
```{r}
genotype_order <- c("WT", "msn2-", "yap1-", "msn4-", "skn7-", "msn4- skn7-")
genotype_color <- RColorBrewer::brewer.pal(8, "Dark2")[c(8, 7, 2, 3, 5, 1)] 
names(genotype_color) <- genotype_order
p.base <- pi.dat %>% 
  filter(new_time == 0, !is.na(new_median)) %>% 
  mutate(Genotype = factor(Genotype, levels = genotype_order)) %>% 
  ggplot(aes(x = Genotype, y = new_median)) +
  geom_point(position = position_jitter(width = 0.1), alpha = 0.5, shape = 16) +
  stat_summary(aes(color = Genotype), fun.data = "mean_cl_boot", geom = "errorbar",
               width = 0.3, linewidth = 1.2) +
  stat_summary(aes(color = Genotype), fun = "mean", geom = "point", size = 4) +
  scale_color_manual(values = genotype_color, guide = "none") +
  labs(y = "Cta1-GFP protein level (a.u.)") +
  theme_cowplot(line_size = 0.7, font_size = 14) +
  background_grid() + panel_border(color = "black", size = 1.5) +
  theme(#legend.position=c(0.05,0.75), 
        #legend.box.background = element_rect(color = "black"),
        #legend.box.margin = margin(3,3,3,3),
        #legend.text = element_text(face = 3), 
        #legend.title = element_text(),
    axis.line = element_blank(),
    axis.title = element_text(size = rel(1)), 
    axis.title.x = element_blank(),
    axis.text.x = element_text(angle = 45, vjust = 1, hjust = 1), axis.text = element_text(size = rel(1)),
  )

p.base
```

Statistical test
```{r}
pi.dat %>% 
  filter(new_time == 0, !is.na(new_median)) %>% 
  mutate(Genotype = factor(Genotype, levels = genotype_order)) %>% 
  lm(new_median ~ Genotype, data = .) %>% 
  summary()
```
> All comparisons are statistically significant at 0.05 level after Bonferroni correction (threshold: 0.0125)

```{r}
# subtract 0 min mean from all time points for each genotype
p.induce <- pi.dat %>% 
  filter(!is.na(new_median)) %>% 
  group_by(Genotype) %>% 
  mutate(delta = new_median - mean(new_median[new_time == 0])) %>% 
  ggplot(aes(x = new_time, y = delta, color = Genotype)) + p.timecourse +
  scale_color_manual(values= genotype_color, limits = names(genotype_color)) +
  scale_x_continuous(breaks = seq(0, 240, by = 30), labels = c(0, '', 60, '', 120, '', 180, '', 240))

plot_grid(p.base, NULL, p.induce, nrow = 1, rel_widths = c(1, 0.1, 1), align = "hv")
#ggsave("output/20221117-Pi-H2O2-timecourse.png", width = 4, height = 3.5)
ggsave("output/20221221-CTA1-trans-delete-base-vs-induce.png", width = 10, height = 4.5)
```

To test for the effect on induction, we can compare the levels at 180 min
```{r}
pi.dat %>% 
  filter(!is.na(new_median)) %>% 
  group_by(Genotype) %>% 
  mutate(delta = new_median - mean(new_median[new_time == 0])) %>% 
  filter(new_time == 120, !is.na(new_median)) %>% 
  mutate(Genotype = factor(Genotype, levels = genotype_order)) %>% 
  lm(new_median ~ Genotype, data = .) %>% 
  summary()
```

#### JY's version
```{r}
genotype_color <- c("#F8766D", "#00BFC4", "#00BA38", "#619CFF", "#E76BF3", "#B79F00") 
# map the color to the genotype variable within pi.dat in its order m2-m4-m4s7-s7-wt-y1
my_genotype <- c("wt", "msn2_ko", "msn4_ko","skn7_ko", "yap1_ko", "msn4skn7_ko")

pi.dat %>%# subset(., !is.na(new_median)) %>%  
  filter(!is.na(new_median)) %>% 
  ggplot(aes(x = new_time, y = new_median, color = Genotype)) + geom_point(size = 0.2) + 
  stat_summary(fun = "mean", geom = "point", size = 2) +
  stat_smooth(aes(x = new_time), se = FALSE) +
  scale_x_continuous(breaks = seq(0,240,30), minor_breaks = seq(0,240,30))+
  theme_bw() +
  theme(legend.position=c(0.15,0.72), legend.background = element_rect(fill = "white", color = "black", size = 1),
        legend.box = "vertical", legend.text = element_text(size=18), 
        legend.title = element_text(size = 18),
        axis.title = element_text(size = 18), 
        axis.title.x = element_blank(),
        axis.text.x = element_text(angle=0,hjust=0.5), axis.text = element_text(size = 18), 
        plot.title = element_text(size = 18, hjust = 0)) + 
  labs(y = "Cta1-GFP protein level (a.u.)") +
  scale_color_manual(values= genotype_color)
  #ggsave(paste0("./output/", gsub("-", "", Sys.Date()), "-trans-TFs-kos.png"),
   #      width = 7, height = 4,units = 'in', dpi = 300)

#library(scales)
#show_col(hue_pal()(5))
#show_col(hue_pal()(6))

pi.dat.summary %>%
  ggplot(aes(x = new_time, y = mean_median, color = Genotype)) + geom_point(size = 0.5) + 
  geom_errorbar(aes(ymin=mean_median-sd, ymax=mean_median+sd), width=3, size = 0.8 ) +
  stat_summary(fun = "mean", geom = "point", size = 2) +
  stat_smooth(aes(x = new_time), se = FALSE) +
  scale_x_continuous(breaks = seq(0,240,30), minor_breaks = seq(0,240,30))+
  theme_bw() +
  theme(legend.position=c(0.18,0.75), legend.background = element_rect(fill = "white", color = "black", size = 1),
        legend.box = "vertical", legend.text = element_text(size=18), 
        legend.title = element_text(size = 18),
        axis.title = element_text(size = 18), 
        axis.title.x = element_blank(),
        axis.text.x = element_text(angle=0,hjust=0.5), axis.text = element_text(size = 18), 
        plot.title = element_text(size = 18, hjust = 0)) + 
  labs(y = "Cta1-GFP protein level (a.u.)") +
  scale_color_manual(values= genotype_color)
  #ggsave(paste0("./output/", gsub("-", "", Sys.Date()), "-sd-trans-TFs-kos.png"),
   #      width = 7, height = 5,units = 'in', dpi = 300)  
  
  # position=position_dodge(0.5)
```
box plot + dot plot for trans TF kos CTA1-GFP
```{r}
pi.tp <- pi.dat %>% 
  filter(Time %in% c("0min", "240min" ) & !is.na(median))

pi.tp %>% 
  mutate(Genotype = factor(Genotype, levels = c("wt", "yap1", "msn2", "msn4", "skn7", "msn4skn7"))) %>% 
  ggplot(aes(x=Genotype, y=new_median, color = Time)) +
  geom_point(aes(x = Genotype, y = new_median), position = position_jitterdodge(jitter.width = 1)) +
  theme_bw() +
  theme(legend.position=c(0.9,0.85), legend.background = element_rect(fill = "white", color = "black", size = 1), legend.box = "vertical", legend.text = element_text(size=18), 
        legend.title = element_text(size = 18),
        axis.title = element_text(size = 18), 
        axis.title.x = element_blank(),
        axis.text.x = element_text(angle=30,hjust = 1, vjust = 1), axis.text = element_text(size = 18), 
        plot.title = element_text(size = 18, hjust = 0)) + 
  stat_summary(aes(group = Time), fun = "median", geom = "point", 
               color = "gray50", position = position_dodge()) + 
  labs(y = "Cta1-GFP protein level (a.u.)") +
  scale_color_manual(values = c("0min" = "#999999", "240min" = "#F564E3"))
  #ggsave(paste0("./output/", gsub("-", "", Sys.Date()), "-TFs-kos-tp.png"),
   #      width = 10, height = 5,units = 'in', dpi = 300)
  

#library(scales)
#show_col(hue_pal()(5))
#show_col(hue_pal()(6))
```


### CTA1 induction under H2O2 [Fig. S?]
```{r}
tmp <- data.cb %>% 
  filter(treat == "H2O2") %>%  
  select(Time = Group, Sample, Plate, replicate = Replicate, genotype, median) %>%
  mutate(Genotype = fct_recode(
    genotype, WT = "wt", `msn2-` = "msn2_ko", `msn4-` = "msn4_ko", 
    `yap1-` = "yap1_ko", `skn7-` = "skn7_ko", `msn4- skn7-` = "msn4skn7_ko",
    `msn2- msn4-` = "msn2msn4_ko", `yap1- skn7-` = "skn7yap1_ko")) %>% 
  mutate(new_median =median/1000) %>% 
  mutate(new_time = as.numeric(gsub("min","",Time)))%>% 
  arrange(new_time, genotype, replicate)
```

plotting
```{r}
genotype_color <- RColorBrewer::brewer.pal(8, "Dark2")[c(8, 7, 2, 3, 5, 1, 4, 6)] 
names(genotype_color) <- c("WT", "msn2-", "yap1-", "msn4-","skn7-", "msn4- skn7-", "yap1- skn7-", "msn2- msn4-")
tmp %>%
  filter(!is.na(new_median)) %>% 
  ggplot(aes(x = new_time, y = new_median, color = Genotype)) + p.timecourse +
  scale_color_manual(values = genotype_color, 
                     limits = c("WT", "msn2-", "msn4-", "msn2- msn4-",
                                "yap1- skn7-", "msn4- skn7-", "yap1-", "skn7-"))
ggsave("output/20221215-trans-deletion-timecourse-H2O2.png", width = 7, height = 5)
```

## Promoter mutants, Msn4 BS
### Prepare data
Import data for 4OL
```{r}
files.OL <- dir("input/OL_csv_merge", pattern = "d01*") # get the file names of the merged
files.OL <- file.path("input/OL_csv_merge", files.OL)         # this appends the directory to the file names

names(files.OL) <- c("Rep1-0Pi","Rep2-0Pi", "Rep3-0Pi")

tp <- c("0min", "30min", "60min", "90min", "120min", "150min",  "180min", "210min", "240min") # ordered time points

# import and process the data to the format we want
## remove characters (0Pi-0min; 240mmin) from group variable 
dat.ord.OL <- #files.OL %>% 
  map_dfr(files.OL, ~read_csv(., na = c("", "NA", "N/A"), col_types = cols()), .id = "Replicate") %>% 
  #bind_rows(.id = "Replicate") %>% 
  filter(`X Parameter` == "BL1-H", is.na(`Y Parameter`)) %>% 
  select(Replicate, Plate = `Experiment`, Group , Sample, Count, median = `X Median`, peak = `X Peak`,
         sd = `X SD`, cv = `X %CV`, rCV = `X %rCV`) %>% 
  mutate_at("Group", str_replace, "-0Pi", "") %>% 
  mutate_at("Group", str_replace, "MIN", "min") %>%
  mutate(Group = ordered(Group, levels = tp)) %>% 
  arrange(Group) 

treat.fil.OL <- read_csv("input/OL_treatment/OL_tr.csv", na = c("","NA","N/A"), col_types = cols()) %>% 
  select(genotype = `group_name`, Sample, treat)

data.OL.tmp <- left_join(dat.ord.OL, treat.fil.OL, by = c("Sample"))
data.OL <- add_row(data.cb,data.OL.tmp)
```

Import data for MutMsn4 #1 & #2
```{r}
files.mutM4 <- dir("input/2_4Imp_gz_merge", pattern = "merge-*") # get the file names of the merged
files.mutM4 <- file.path("input/2_4Imp_gz_merge", files.mutM4)         # this appends the directory to the file names

names(files.mutM4) <- c("Rep1-H2O2", "Rep1-0Pi","Rep2-0Pi", "Rep3-0Pi" ,"Rep2-H2O2", "Rep3-H2O2")

tp <- c("0min", "30min", "60min", "90min", "120min", "150min",  "180min", "210min", "240min") # ordered time points

# import and process the data to the format we want
## remove characters (0Pi-0min; 240mmin) from group variable 
dat.ord.mutM4 <- #files.mutM4 %>% 
  map_dfr(files.mutM4, ~read_csv(., na = c("", "NA", "N/A"), col_types = cols()), .id = "Replicate") %>% 
  #bind_rows(.id = "Replicate") %>% 
  filter(`X Parameter` == "BL1-H", is.na(`Y Parameter`)) %>% 
  select(Replicate, Plate, Group, Sample, Count, median = `X Median`, peak = `X Peak`,
         sd = `X SD`, cv = `X %CV`, rCV = `X %rCV`) %>% 
  mutate_at("Group", str_replace, "0Pi-", "") %>% 
  mutate_at("Group", str_replace, "240m", "240") %>%
  mutate(Group = ordered(Group, levels = tp)) %>% 
  arrange(Group)   
```

Subset the dataset and merge with the genotype information
```{r}
# get genotype info and petites info
geno <- read.csv("input/Genotype_2_4Imp.csv")
names(geno) <- c("HB_strain", names(geno)[-1])

# get treatment info
tr_files.mutM4 <- dir("input/2_4Imp_treatment", pattern = "d0*")
tr_files.mutM4 <- file.path("input/2_4Imp_treatment", tr_files.mutM4)
names(tr_files.mutM4) <-  c("Rep1-H2O2", "Rep1-0Pi","Rep2-0Pi", "Rep3-0Pi" ,"Rep2-H2O2", "Rep3-H2O2")

#sum(treat.fil$treat == "0Pi")
#sum(treat.fil$treat == "H2O2")

# combine HB_strain info to rule out petites
treat.fil.mutM4 <- #tr_files.mutM4 %>% 
  map_dfr(tr_files.mutM4, ~read_csv(., na = c("","NA","N/A"), col_types = cols()), .id = "Replicate") %>% 
  #bind_rows(.id = "Replicate") %>% 
  filter(!is.na(`group_name`)) %>% 
  select(Replicate, genotype = `group_name`, Sample, treat) %>% 
  left_join(., geno, by = c("genotype", "Sample")) # %>% 
  #subset(!is.na(YPG))


# combine genotype and treat variables into the flow data
data.mutM4 <- left_join(dat.ord.mutM4, treat.fil.mutM4, by = c("Replicate", "Sample")) %>% 
  filter(!is.na(YPG)) %>% 
  select(Replicate,Plate,Group,Sample,genotype,median,treat)
data.OL_mutM4 <- add_row(data.OL,data.mutM4)

# select out 0Pi data
pi.dat.OL_mutM4 <- data.OL_mutM4 %>% 
  filter(treat == "0Pi" & genotype %in% c("wt","MutMsn4 #1","MutMsn4 #2","msn4_ko","4OL")) %>%  
  select(Time = Group, Sample, Plate, replicate = Replicate, genotype, median) %>%
  mutate(Genotype = fct_recode(genotype, WT = "wt", `msn4-` = "msn4_ko", 
                               `muBS#1` = "MutMsn4 #1", `muBS#2` = "MutMsn4 #2")) %>% 
  mutate(new_median =median/1000) %>% 
  mutate(new_time = as.numeric(gsub("min","",Time)))%>% 
  arrange(new_time, genotype, replicate)

pi.dat.OL_mutM4.summary <- pi.dat.OL_mutM4 %>% 
  filter(!is.na(new_median)) %>%
  group_by(Genotype,Time) %>% 
  summarise(sd = sd(new_median, na.rm = TRUE), mean_median = mean(new_median)) %>% 
  mutate(.,`Time (min)`= as.numeric(gsub("min","",Time)))
```
### Bin version

**Update 2022-11-08**
```{r}
MutM4_color <- RColorBrewer::brewer.pal(8, "Dark2")[c(8, 3, 7, 6, 4)] 
names(MutM4_color) <- c("WT", "msn4-", "muBS#1","muBS#2", "4OL")
MutM4_color["muBS#1"] = "#e6d602ff" # to avoid having the same color as the msn2-

pi.dat.OL_mutM4 %>%
  filter(!is.na(new_median)) %>% 
  ggplot(aes(x = new_time, y = new_median, color = Genotype)) + p.timecourse +
  scale_color_manual(values= MutM4_color) +
  scale_y_continuous(limits = c(NA, 40))
#ggsave("output/20221108-msn4BS-cis-deletion-timecourse.png", width = 4.5, height = 4)
```
### JY version
```{r}
MutM4_color <- c("#B79F00", "#00BFC4",  "#F8766D", "#00BA38", "#F564E3")

pi.dat.OL_mutM4 %>% 
  subset(., !is.na(new_median)) %>%  
  ggplot(aes(x = new_time, y = new_median, color = Genotype)) + geom_point(size = 0.2) + 
  stat_summary(fun = "mean", geom = "point", size = 2) +
  stat_smooth(aes(x = new_time), se = FALSE) +
  scale_x_continuous(breaks = seq(0,240,30), minor_breaks = seq(0,240,30))+
  theme_bw() +
  theme(legend.position=c(0.18,0.75), legend.background = element_rect(fill = "white", color = "black", size = 1),
        legend.box = "vertical", legend.text = element_text(size=18), 
        legend.title = element_text(size = 18),
        axis.title = element_text(size = 18), 
        axis.title.y = element_text(size = 18),
        axis.title.x = element_blank(),
        axis.text.x = element_text(angle=0,hjust=0.5), axis.text = element_text(size = 18), 
        plot.title = element_text(size = 18, hjust = 0)) + 
  labs(y = "Cta1-GFP protein level (a.u.)") +
  scale_color_manual(values= MutM4_color)
  #ggsave(paste0("./output/", gsub("-", "", Sys.Date()), "-cis-Msn4BS.png"),
   #      width = 7, height = 4,units = 'in', dpi = 300)

pi.dat.OL_mutM4.summary %>%
  ggplot(aes(x = `Time (min)`, y = mean_median, color = Genotype)) + geom_point(size = 0.5) + 
  geom_errorbar(aes(ymin=mean_median-sd, ymax=mean_median+sd), width=3, size = 0.8 ) +
  stat_summary(fun = "mean", geom = "point", size = 2) +
  stat_smooth(aes(x = `Time (min)`), se = FALSE) +
  scale_x_continuous(breaks = seq(0,240,30), minor_breaks = seq(0,240,30))+
  theme_bw() +
  theme(legend.position=c(0.18,0.75), legend.background = element_rect(fill = "white", color = "black", size = 1),
        legend.box = "vertical", legend.text = element_text(size=18), 
        legend.title = element_text(size = 18),
        axis.title = element_text(size = 18), 
        axis.title.x = element_text(size = 18),
        axis.text.x = element_text(angle=0,hjust=0.5), axis.text = element_text(size = 18), 
        plot.title = element_text(size = 18, hjust = 0)) + 
  labs(y = "Cta1-GFP protein level (a.u.)") +
  scale_color_manual(values= MutM4_color)
  #ggsave(paste0("./output/", gsub("-", "", Sys.Date()), "-sd-cis-Msn4BS-kos.png"),
   #      width = 7, height = 5,units = 'in', dpi = 300) 

  
#library(scales)
#show_col(hue_pal()(5))
#show_col(hue_pal()(6))
```

box plot + dot plot for trans TF kos CTA1-GFP
```{r}
pi.MutM4_OL.tp <- pi.dat.OL_mutM4 %>% 
  filter(Time %in% c("0min", "240min" ) & !is.na(median))


pi.MutM4_OL.tp %>% 
  mutate(Genotype = factor(Genotype, levels = c("wt", "msn4", "4OL", "MutMsn4 #1", "MutMsn4 #2"))) %>% 
  ggplot(aes(x=Genotype, y=new_median, color = Time)) +
  geom_point(aes(x = Genotype, y = new_median), position = position_jitterdodge(jitter.width = 1)) +
  theme_bw() +
  theme(legend.position=c(0.4,0.85), legend.background = element_rect(fill = "white", color = "black", size = 1), legend.box = "vertical", legend.text = element_text(size=18), 
        legend.title = element_text(size = 18),
        axis.title = element_text(size = 18), 
        axis.title.x = element_blank(),
        axis.text.x = element_text(angle=30,hjust = 1, vjust = 1), axis.text = element_text(size = 18), 
        plot.title = element_text(size = 18, hjust = 0)) + 
  stat_summary(aes(group = Time), fun = "median", geom = "point", 
               color = "gray50", position = position_dodge()) + 
  labs(y = "Cta1-GFP protein level (a.u.)") +
  scale_color_manual(values = c("0min" = "#999999", "240min" = "#F564E3"))
  #ggsave(paste0("./output/", gsub("-", "", Sys.Date()), "-cis-2-4Imp-tp.png"),
   #      width = 7, height = 5,units = 'in', dpi = 300)
  
           
```

## Promoter variants, Skn7
### Prepare data
Import data
```{r}
files.mutS7 <- dir("input/6_8Imp_gz_merge", pattern = "merge-*") # get the file names of the merged
files.mutS7 <- file.path("input/6_8Imp_gz_merge", files.mutS7)         # this appends the directory to the file names

names(files.mutS7) <- c("Rep1-H2O2", "Rep1-0Pi","Rep2-0Pi", "Rep3-0Pi" ,"Rep2-H2O2", "Rep3-H2O2")

tp <- c("0min", "30min", "60min", "90min", "120min", "150min",  "180min", "210min", "240min") # ordered time points

# import and process the data to the format we want
## remove characters (0Pi-0min; 240mmin) from group variable 
dat.ord.mutS7 <- #files.mutS7 %>% 
  map_dfr(files.mutS7, ~read_csv(., na = c("", "NA", "N/A"), col_types = cols()), .id = "Replicate") %>% 
  #bind_rows(.id = "Replicate") %>% 
  filter(`X Parameter` == "BL1-H", is.na(`Y Parameter`)) %>% 
  select(Replicate, Plate, Group, Sample, Count, median = `X Median`, peak = `X Peak`,
         sd = `X SD`, cv = `X %CV`, rCV = `X %rCV`) %>% 
  mutate_at("Group", str_replace, "-H2O2", "") %>% 
  mutate_at("Group", str_replace, "MIN", "min") %>%
  mutate(Group = ordered(Group, levels = tp)) %>% 
  arrange(Group)   
```

Subset the dataset and merge with the genotype information
```{r}
# get genotype info and petites info
geno.mutS7 <- read.csv("input/Genotype_6_8Imp.csv")
names(geno.mutS7) <- c("HB_strain", names(geno)[-1])

# get treatment info
tr_files.mutS7 <- dir("input/6_8Imp_treatment", pattern = "d0*")
tr_files.mutS7 <- file.path("input/6_8Imp_treatment", tr_files.mutS7)
names(tr_files.mutS7) <-  c("Rep1-H2O2", "Rep1-0Pi","Rep2-0Pi", "Rep3-0Pi" ,"Rep2-H2O2", "Rep3-H2O2")


# combine HB_strain info to rule out petites
treat.fil.mutS7 <- #tr_files.mutS7 %>% 
  map_dfr(tr_files.mutS7, ~read_csv(., na = c("","NA","N/A"), col_types = cols()), .id = "Replicate") %>%
  #bind_rows(.id = "Replicate") %>% 
  filter(!is.na(`group_name`)) %>% 
  select(Replicate, genotype = `group_name`, Sample, treat) %>% 
  left_join(geno.mutS7, by = c("genotype", "Sample")) # %>% 
  #subset(!is.na(YPG))


# combine genotype and treat variables into the flow data
data.mutS7 <- left_join(dat.ord.mutS7, treat.fil.mutS7, by = c("Replicate", "Sample")) %>% 
  filter(!is.na(YPG)) %>% 
  select(Replicate,Plate,Group,Sample,genotype,median,treat)

data.OL_mutS7 <- add_row(data.OL,data.mutS7)


# select out 0Pi data
pi.dat.OL_mutS7 <- data.OL_mutS7 %>% 
  filter(treat == "0Pi" & genotype %in% c("wt","MutS7#1","MutS7#2","8OL","skn7_ko")) %>%  
  select(Time = Group, Sample, Plate, replicate = Replicate, genotype, median) %>%
  mutate(Genotype = fct_recode(genotype, WT = "wt", `skn7-` = "skn7_ko",
                               `muBS#1` = "MutS7#1", `muBS#2` = "MutS7#2")) %>% 
  mutate(new_median =median/1000) %>% 
  mutate(new_time = as.numeric(gsub("min","",Time)))%>% 
  arrange(new_time, genotype, replicate)

pi.dat.OL_mutS7.summary <- pi.dat.OL_mutS7 %>% 
  filter(!is.na(new_median)) %>%
  group_by(Genotype, Time) %>% 
  summarise(sd = sd(new_median, na.rm = TRUE), mean_median = mean(new_median)) %>% 
  mutate(`Time (min)`= as.numeric(gsub("min","",Time)))
```

### Bin's version
```{r}
MutS7_color <- RColorBrewer::brewer.pal(8, "Dark2")[c(8, 5, 7, 6, 4)] 
names(MutS7_color) <- c("WT", "skn7-", "muBS#1","muBS#2", "8OL")
MutM4_color["muBS#1"] = "#e6d602ff" # to avoid having the same color as the msn2-

pi.dat.OL_mutS7 %>%
  filter(!is.na(new_median)) %>% 
  ggplot(aes(x = new_time, y = new_median, color = Genotype)) + p.timecourse +
  scale_color_manual(values= MutS7_color) +
  scale_y_continuous(limits = c(NA, 40))
#ggsave("output/20221108-skn7BS-cis-deletion-timecourse.png", width = 4.5, height = 4)
```

### JY's version
```{r}
MutS7_color <- c("#B79F00", "#00BFC4",  "#F8766D", "#619CFF", "#F564E3")

pi.dat.OL_mutS7 %>% 
  subset(., !is.na(new_median)) %>%  
  ggplot(aes(x = new_time, y = new_median, color = Genotype)) + geom_point(size = 0.2) + 
  stat_summary(fun = "mean", geom = "point", size = 2) +
  stat_smooth(aes(x = new_time), se = FALSE) +
  scale_x_continuous(breaks = seq(0,240,30), minor_breaks = seq(0,240,30))+
  theme_bw() +
  theme(legend.position="right", legend.box = "vertical", legend.text = element_text(size=18), 
        legend.title = element_text(size = 18),
        axis.title = element_text(size = 18), 
        axis.title.y = element_blank(),
        axis.title.x = element_blank(),
        axis.text.x = element_text(angle=0,hjust=0.5), axis.text = element_text(size = 18), 
        plot.title = element_text(size = 18, hjust = 0)) + 
  labs(y = "Cta1-GFP protein level (a.u.)") +
  scale_color_manual(values= MutS7_color)
  #ggsave(paste0("./output/", gsub("-", "", Sys.Date()), "-cis-Skn7BS.png"),
   #      width = 7, height = 4,units = 'in', dpi = 300)

pi.dat.OL_mutS7.summary %>%
  ggplot(aes(x = `Time (min)`, y = mean_median, color = Genotype)) + geom_point(size = 0.5) + 
  geom_errorbar(aes(ymin=mean_median-sd, ymax=mean_median+sd), width=3, size = 0.8 ) +
  stat_summary(fun = "mean", geom = "point", size = 2) +
  stat_smooth(aes(x = `Time (min)`), se = FALSE) +
  scale_x_continuous(breaks = seq(0,240,30), minor_breaks = seq(0,240,30))+
  theme_bw() +
  theme(legend.position=c(0.18,0.75), legend.background = element_rect(fill = "white", color = "black", size = 1),
        legend.box = "vertical", legend.text = element_text(size=18), 
        legend.title = element_text(size = 18),
        axis.title = element_text(size = 18), 
        axis.title.x = element_text(size = 18),
        axis.text.x = element_text(angle=0,hjust=0.5), axis.text = element_text(size = 18), 
        plot.title = element_text(size = 18, hjust = 0)) + 
  labs(y = "Cta1-GFP protein level (a.u.)") +
  scale_color_manual(values= MutS7_color)
  #ggsave(paste0("./output/", gsub("-", "", Sys.Date()), "-sd-cis-Skn7BS.png"),
   #      width = 7, height = 5,units = 'in', dpi = 300) 


  
#library(scales)
#show_col(hue_pal()(5))
#show_col(hue_pal()(6))
```

```{r}
pi.MutS7_OL.tp <- pi.dat.OL_mutS7 %>% 
  filter(Time %in% c("0min", "240min" ) & !is.na(median))


pi.MutS7_OL.tp %>% 
  mutate(Genotype = factor(Genotype, levels = c("wt", "skn7", "8OL", "MutS7#1", "MutS7#2"))) %>% 
  ggplot(aes(x=Genotype, y=new_median, color = Time)) +
  geom_point(aes(x = Genotype, y = new_median), position = position_jitterdodge(jitter.width = 1)) +
  theme_bw() +
  theme(legend.position=c(0.4,0.85), legend.background = element_rect(fill = "white", color = "black", size = 1), legend.box = "vertical", legend.text = element_text(size=18), 
        legend.title = element_text(size = 18),
        axis.title = element_text(size = 18), 
        axis.title.x = element_blank(),
        axis.text.x = element_text(angle=30,hjust = 1, vjust = 1), axis.text = element_text(size = 18), 
        plot.title = element_text(size = 18, hjust = 0)) + 
  stat_summary(aes(group = Time), fun = "median", geom = "point", 
               color = "gray50", position = position_dodge()) + 
  labs(y = "Cta1-GFP protein level (a.u.)") +
  scale_color_manual(values = c("0min" = "#999999", "240min" = "#F564E3"))
  #ggsave(paste0("./output/", gsub("-", "", Sys.Date()), "-cis-6-8Imp-tp.png"),
   #      width = 7, height = 5,units = 'in', dpi = 300)
```

## Combine Skn7 & Msn4 BS mutant data
```{r}
data.OL_cb <- add_row(data.OL_mutM4,data.mutS7)

pi.dat.OL_cb <- data.OL_cb %>% 
  filter(treat == "0Pi" & genotype %in% c("wt","MutS7#1","MutS7#2","8OL","skn7_ko","MutMsn4 #1","MutMsn4 #2","msn4_ko","4OL")) %>%  
  select(Time = Group, Sample, Plate, replicate = Replicate, genotype, median) %>%
  mutate(.,Genotype = gsub("_ko","",genotype)) %>% 
  mutate(.,new_median =median/1000) %>% 
  mutate(.,new_time = as.numeric(gsub("min","",Time)))%>% 
  arrange(new_time, genotype, replicate)

pi.OL.tp <- pi.dat.OL_cb %>% 
  filter(Time %in% c("0min", "240min" ) & !is.na(median))


pi.OL.tp %>% 
  mutate(Genotype = factor(Genotype, levels = c("wt","4OL","msn4","MutMsn4 #1","MutMsn4 #2", "8OL", "skn7","MutS7#1", "MutS7#2"))) %>% 
  ggplot(aes(x=Genotype, y=new_median, color = Time)) +
  geom_point(aes(x = Genotype, y = new_median), position = position_jitterdodge(jitter.width = 1)) +
  theme_bw() +
  theme(legend.position = "none",
        axis.title = element_text(size = 18), 
        axis.title.x = element_blank(),
        axis.text.x = element_text(angle=30,hjust = 1, vjust = 1), axis.text = element_text(size = 18), 
        plot.title = element_text(size = 18, hjust = 0)) + 
  stat_summary(aes(group = Time), fun = "median", geom = "point", 
               color = "gray50", position = position_dodge()) + 
  labs(y = "Cta1-GFP protein level (a.u.)") +
  scale_color_manual(values = c("0min" = "#999999", "240min" = "#F564E3"))
 # ggsave(paste0("./output/", gsub("-", "", Sys.Date()), "-cis-all-tp.png"),
  #       width = 10, height = 5,units = 'in', dpi = 300)
           
```

## Rim15∆
### Prepare data
```{r}
files.r15 <- dir("input/rim15_gz_merge", pattern = "merge-*", full.names = TRUE) # get the file names of the merged
names(files.r15) <- c("Rep1-0Pi-Ctrl","Rep2-0Pi-Ctrl", "Rep3-0Pi-Ctrl") # correspond to treatment files

tp <- c("0min",  "30min", "60min", "90min", "120min", "150min",  "180min", "210min", "240min") # ordered time points

# import and process the data to the format we want
dat.ord.r15 <-  map_dfr(files.r15, ~read_csv(., na = c("", "NA", "N/A"), col_types = cols()), .id = "Replicate") %>% 
  #bind_rows(.id = "Replicate") %>% 
  filter(`X Parameter` == "BL1-H", is.na(`Y Parameter`)) %>% 
  select(Replicate, Plate, Group, Sample, Count, median = `X Median`, peak = `X Peak`,
         sd = `X SD`, cv = `X %CV`, rCV = `X %rCV`) %>% 
  mutate(Group = ordered(Group, levels = tp)) %>% 
  arrange(Group)
#write_csv(dat.ord, file = "input/rim15_gz_merge/2021014-1019-rim15-KO-flow-data.csv")
```


```{r}
# get genotype info
tr_files.r15 <- dir("input/rim15_treatment", pattern = "d0*", full.names = TRUE)
#tr_files.r15 <- file.path("input/rim15_treatment", tr_files.r15)
names(tr_files.r15) <- c("Rep1-0Pi-Ctrl","Rep2-0Pi-Ctrl", "Rep3-0Pi-Ctrl")


treat.fil.r15 <-  map_dfr(tr_files.r15, ~read_csv(., na = c("","NA","N/A"), col_types = cols()), .id = "Replicate") %>% 
  #bind_rows(.id = "Replicate") %>% 
  filter(!is.na(`group_name`)) %>% 
  select(Replicate, genotype = `group_name`, Sample, treat)

# combine genotype and treat variables into the flow data
data.r15 <- left_join(dat.ord.r15, treat.fil.r15, by = c("Replicate", "Sample"))
#sum(is.na(data$Group))

#write_csv(data, file = "gz_merge/20210724-0805-treat-gt-Rim15-KO-flow-data.csv.gz")

# select out 0Pi data
pi.dat.r15 <- data.r15 %>% 
  filter(treat == "0Pi" & genotype %in% c("wt","msn4_nat_ko","rim15_ko","msn4rim15_ko")) %>%  
  select(Time = Group, Sample, Plate, replicate = Replicate, genotype, median) %>%
  mutate(Genotype = fct_recode(genotype, WT = "wt", `msn4-` = "msn4_nat_ko", `rim15-` = "rim15_ko", `msn4- rim15-` = "msn4rim15_ko"),
         new_median = median/1000,
         new_time = as.numeric(gsub("min","",Time))) %>% 
  #mutate(Genotype = gsub("_nat","",Genotype)) %>% 
  arrange(new_time, genotype, replicate) 
```

### Bin's version

```{r}
genotype_color <- RColorBrewer::brewer.pal(8, "Dark2")[c(8, 3, 7, 5, 2, 1)] 
names(genotype_color) <- c("WT", "msn4-", "rim15-", "msn4- rim15-")

pi.dat.r15 %>%
  filter(!is.na(new_median)) %>% 
  ggplot(aes(x = new_time, y = new_median, color = Genotype)) + p.timecourse +
  scale_color_manual(values= genotype_color) +
  scale_
  theme_cowplot(font_size = 16, line_size = 0.7) +
  theme(legend.position=c(0.05,0.75),
        #legend.box.background = element_rect(color = "black"),
        #legend.box.margin = margin(3,3,3,3),
        legend.text = element_text(face = 3), 
        legend.title = element_text(),
        axis.title = element_text(size = rel(1)), 
        axis.title.x = element_blank(),
        axis.text.x = element_text(hjust=0.5), axis.text = element_text(size = rel(1)),
  )

ggsave("output/20221118-rim15-deletion-timecourse.png", width = 4.5, height = 4)
```