---
title: "Plot the expression of Pi starvaiton induced genes relevant for oxidative stress, in _C. glabrata_"
author: "Bin He"
date: "2022-04-28 (updated `r Sys.Date()`)"
output: 
  html_notebook:
    toc: true
    toc_float: true
    code_folding: hide
---

```{r load_library, echo=FALSE}
# install the package if not already available
suppressPackageStartupMessages(library(tidyverse))
suppressPackageStartupMessages(library(cowplot))
suppressPackageStartupMessages(library(ggtext))
suppressPackageStartupMessages(library(edgeR))
suppressPackageStartupMessages(library(SummarizedExperiment))
suppressPackageStartupMessages(library(pheatmap))
suppressPackageStartupMessages(require(broom))
```

## Goal
A recent paper <https://doi.org/10.1101/2021.09.28.462173> identified genes induced in _C. glabrata_ after being engulfed by macrophages. We are interested in knowning if the gene sets identified there were also induced during phosphate starvation. Conversely, we are also interested in knowing if the PHO genes were induced after _C. glabrata_ cells were phagocytosed.

## Data
### _C. glabrata_ time course in macrophage
This data is kindly provided by Dr. Chris Koon's lab at the University of Macao. A student in his group, Lan Qing, compiled the data and sent us a PPT file that contained both a matrix and some plots. See the paper linked above by DOI for method details. Briefly, _C. glabrata_ culture was added to human THP-1 differentiated monocyte cells at an MOI of 5:1, and sampled at 0.5h, 2h, 4h, 6h and 8h post infection time points. RNA PolII ChIP-seq was performed in place of RNA-seq. Bioinformatic analyses involved mapping quality-checked short reads to the genome using Bowtie2, feeding the resulting BAM files to MACS2 and calculating a normalized read count (RPKM) for each gene.

My understanding is that they didn't include a pre-infection time point and thus we will rely on the time course trend to infer whether a gene is "induced" during infection.
```{r}
chip.phage <- read_tsv("../input/Cgla-mphage-Koon2021/20230428-Koon-lab-mphage-PHO-genes-Cgla.tsv", col_types = cols())
```

### _C. glabrata_ -Pi time course
Load normalized gene counts for the -Pi time course in _C. glabrata_. Data collected in 2017 by Bin He. Pre-processing was done as described under the E009/seq-results folder.

Below is a simple chart to show the design:

| Genotype | Time point  | Replicates | Comment |
| -------- | :---------- | :--------: | ------- |
| 80∆      | rich media  | 2          | for consistency check with my previous data |
| 80∆ 4∆   | rich media  | 2          | for consistency check with my previous data |
| wt-1     | pre, 20', 30', 45', 60', 90', 120', 150', 180', 240' | 1 | time course for wt  |
| 4∆-1     | pre, 20', 30', 45', 60', 90', 120', 150', 180', 240' | 1 | time course for 4∆  | 
| wt-2     | pre, 20', 30', 45', 60',      120',       180', 240' | 1 | biol. repl. for wt-1|
| 4∆-2     | pre, 20', 30', 45', 60',      120',       180'       | 1 | biol. repl. for 4∆-1| 

Count matrix
```{r}
# raw data count
tmp <- read_table("../input/Cgla-Pi-He2017-unpub/20171025_reads_per_transcript_filtered.txt",
                col_types = cols()) %>% 
  column_to_rownames(var = "gene.names")
# sample sheet
sample <- 
  read_csv("../input/Cgla-Pi-He2017-unpub/20171019-Ex009-experiment-setup.csv",
           col_types = cols()) %>%
  mutate(
    Timepoint = factor(Timepoint, levels = 
                         c("pre", "20m", "30m", "45m", "60m", "90m", 
                           "120m", "150m", "180m", "240m", "del80")),
    Group = paste(Genotype, Timepoint, sep = ".")
  )
# rearrange the columns in `raw` according to the samples order in `sample`
tmp <- tmp[, match(sample$Sample, colnames(tmp))]
```

annotation
```{r}
anno <- read_tsv("../input/annotation/C_glabrata_gene_for_mapping_s02-m07-r04.bed",
                 col_names =  c("Chr", "Start", "End", "GeneID", "Not.Use",
                                "Strand", "GeneName", "Type.of.Gene")) %>% 
                   select(Chr, GeneID, GeneName, Type.of.Gene)
cgToSc <- read_tsv("../input/annotation/C_glabrata_CBS138_S_cerevisiae_orthologs.txt", 
                   comment = "#",
                   col_names = c("cgid","cgname","cgid2","scid","scname","scid2"))
```

```{r}
cgToSc <- cgToSc %>% 
  select(CglaID = cgid, CglaName = cgname, ScerID = scid, ScerName = scname)
anno <- left_join(anno, cgToSc, by = c("GeneID" = "CglaID")) %>% 
  select(CglaID = GeneID, CglaName, ScerID, ScerName, Chr, Type.of.Gene)
```

Construct DGEList for normalization and calculating count per million
```{r}
# organize gene annotations
anno.select <- column_to_rownames(anno, "CglaID")[match(rownames(tmp), anno$CglaID),]
all(rownames(anno.select) == rownames(tmp))
# sample sheet
all(sample$Sample == names(tmp))
# construct DGEList
cgla.eds <- DGEList(
  counts = tmp, 
  samples = column_to_rownames(sample, "Sample"), 
  genes = anno.select
)
cgla.eds <- calcNormFactors(cgla.eds, method = "TMM")
lcpm <- cpm(cgla.eds, normalized.lib.sizes = TRUE, log = TRUE)
```


Construct a SummarizedExperiment object
```{r}
se0 <- SummarizedExperiment(
  assays = lcpm, 
  rowData = anno.select,
  colData = column_to_rownames(sample, "Sample")
  )
# subset the experiment to include just the wt time course
se <- se0[, se0$Genotype == "wt" & se0$Timepoint != "del80"]
```

### _S. cerevisiae_ -Pi 1hr gene expression
Description and code below copied/adapted from `uncat-analyses/CTA1-induction-noPi`

_Data source_

| GEO# | Description | Reference |
| ---- | ----------- | --------- |
| GSE23580 | Microarray expression analyses of _S. cerevisiae_ wt, _pho4∆_, _pho2∆_ in rich or no Pi media, sampled at 1hr | Zhou & O'Shea 2011 |

    Zhou X, O’Shea EK. 2011. Integrated approaches reveal determinants of genome-wide binding and function of the transcription factor Pho4. Mol Cell 42:826–836.

The analysis below is inspired by a GEOquery workshop offered by Jason Ratcliff at the Iowa Institute of Human Genetics. `getGEO()` will return an ExpressionSet object, which is then converted into a "SummarizedExperiment" object, which is a more modern data structure that is easier to deal with.
```{r}
#Sys.setenv("VROOM_CONNECTION_SIZE" = 131072*10) # increase the local cache size
gse <- GEOquery::getGEO(filename = "../input/Scer-Pi-Zhou2011/Zhou2011/GSE23580_series_matrix.txt.gz") %>% 
  as("SummarizedExperiment")
```

The experimental information is stored in the `colData` fields. The ones we are interested in are:
```{r}
colData(gse) %>% 
  as_tibble() %>%
  filter(grepl("Wild type no vs high Pi conditions", title) | grepl("Comparison 1$", title)) %>% 
  select(title, geo_accession, organism = organism_ch1, strain = characteristics_ch1, 
         condition_ch1 = characteristics_ch1.2, label_ch1, condition_ch2 = characteristics_ch2.2, label_ch2)
```

The first three are part of the mutant cycle while the latter four are not. The latter four are said to have incorporated a dye swap, although I can't tell how the swap was done from the table above.

Separately extract the two subsets and examine them separately.
```{r}
gse1 <- gse[, grepl("Comparison 1$", gse$title)]
gse2 <- gse[, grepl("Wild type no vs high Pi conditions", gse$title)]
```

### Gene list for macrophage experiment
I manually copied the gene names from Fig 1 - supplementary figures 2 and 3. The figures showed either gene IDs or gene names. I used [CGD](candidagenome.org) to translate the gene names to IDs, and then obtained annotation information for all 93 genes (94 rows, one gene redundant, belonging to two groups). I merged the above information in R. The process was a bit messy and not recorded here. But the original manually entered data was preserved as `c-glabrata-macrophage-induced-raw.csv` in the `input/gene-list` folder. The merged dataset is in the same folder.
```{r}
# manually compiled datasets
cgla.phage <- read_csv("../input/gene-list/c-glabrata-macrophage-induced.csv", col_types = cols()) %>%
  mutate(geneName = str_to_upper(geneName),
         category = fct_recode(category, "cell cycle arrest" = "cell cycle arrest in response to nitrogen starvation"))
```

### Mapping between the two genomes

Mapping between the two species are downloaded from [CGD](www.candidagenome.org). Two types of mappings are available: [orthology](http://www.candidagenome.org/download/homology/orthologs/) is based on the [Yeast Gene Order Browser](http://ygob.ucd.ie/); [best hits](http://www.candidagenome.org/download/homology/best_hits/) is based on `blastp` and is only performed for genes lacking a credible ortholog. Note that because both mappings start from _C. glabrata_, they are not the ideal mapping to use for identifying the _C. glabrata_ ortholog for _S. cerevisiae_ genes, but should be sufficient for the current purpose. This is because most of the genes do have 1-to-1 orthologs between the two species and thus starting from either species should yield the same mapping.

Below our goal is to reverse-engineer an scToCg one-to-one mapping from the two cgToSc mappings.
```{r}
# cgToSc is loaded already and stores the orthology mapping
# read the best hit mapping
cgToSc.bh <- read_tsv("../input/annotation/C_glabrata_CBS138_S_cerevisiae_best_hits.txt", 
                           comment = "#", col_types = cols(), 
                           col_names = c("CglaID","CglaName","Cgla.id","ScerID","ScerName","Scer.id")) %>% 
  select(CglaID, CglaName, ScerID, ScerName)
re.scToCg <- bind_rows(
  ortholog = cgToSc, 
  # only include the rows in the best hit table if the Scer gene is not already in the orthology mapping
  # for Scer genes with more than one Cgla genes mapped to it by blastp, randomly pick the first one
  `best hit` = cgToSc.bh %>% filter(!ScerID %in% cgToSc$ScerID) %>% group_by(ScerID) %>% filter(row_number() == 1),
  .id = "mapping")
```

Combine the orthology and best hits mapping
```{r}
cgla.list <- cgla.phage %>% 
  dplyr::rename(CglaID = geneID, CglaName = geneName) %>%
  left_join(cgToSc %>% select(CglaID, CglaName, ScerID.or = ScerID, ScerName.or = ScerName)) %>% 
  # note that the following is not a robust procedure. it just happens that none of the Scer genes in 
  # scer.osr have more than one best hit mapped Cgla genes
  left_join(cgToSc.bh %>% select(CglaID, CglaName, ScerID.bh = ScerID, ScerName.bh = ScerName)) %>% 
  mutate(
    ScerID = ifelse(is.na(ScerID.or), ScerID.bh, ScerID.or),
    ScerName = ifelse(is.na(ScerName.or), ScerName.bh, ScerName.or),
    name = ifelse(is.na(CglaName), ScerName, CglaName)
  ) %>% 
  select(CglaID, CglaName, ScerID, ScerName, name, everything())
```


## Analysis
### Plotting _C. glabrata_ -Pi time course for genes induced during infection
```{r}
myGenesPlot <- function(genes = "CAGL0B02475g", names = "PMU2") {
  # this function takes in the read count matrix (normalized and transformed) and a list of gene IDs
  # and plots the values stratified by genotype and timepoint
  if(!is.null(names(genes))){
    names = names(genes)
  }
  else if(length(genes) != length(names)){
    stop("length of 'genes' must equal length of 'names'")
  }
  
  if(any(!genes %in% rownames(se))){
    print("The following genes are not included in the experiment. Check to make sure that the gene names are correct.")
    print(setdiff(genes, rownames(se)))
    names = names[genes %in% rownames(se)]
    genes = intersect(genes, rownames(se))
  }
  
  # construct tibble for plotting
  dat <- t(assay(se[genes,])) %>% as_tibble(rownames = NA) %>% rownames_to_column(var = "sample") # subset the expression matrix
  tb <- colData(se) %>% as_tibble(rownames = NA) %>% 
    rownames_to_column(var = "sample") %>% 
    select(sample, Genotype, Timepoint) %>%
    left_join(dat, by = "sample") %>% 
    pivot_longer(cols = starts_with("CAGL"), names_to = "gene", values_to = "log2 cpm") %>% 
    left_join(tibble(gene = genes, name = names), by = "gene")
    
  p <- ggplot(tb, aes(x = Timepoint, y = `log2 cpm`)) + 
    geom_point() + geom_smooth(method = "loess", formula = y ~ x, aes(x = as.numeric(Timepoint), y = `log2 cpm`)) + 
    xlab("") + ylab("log2 count per million") + facet_wrap(~paste(gene, name, sep = ":"), scale = "free_y") +
    theme_bw(base_size = 12) + 
    theme(axis.text.x = element_text(angle = 45, size = rel(0.75), hjust = 1),
          axis.text.y = element_text(size = rel(0.75)),
          strip.text = element_text(size = rel(0.75)))
  return(p)
}
```

```{r fig.show=FALSE}
group <- levels(cgla.list$category)
group
categories <- lapply(group, function(x){
  test <- cgla.list %>% 
    filter(category == x) %>% 
    select(name, CglaID) %>% 
    filter(!is.na(CglaID)) %>% 
    deframe()
})
names(categories) <- group
for (i in seq_along(categories)){
  sprintf("Plotting for %s", categories[[i]])
  p <- myGenesPlot(categories[[i]])
  print(p + ggtitle(str_to_upper(group[i])) + theme(plot.title = element_text(hjust = 0.5)))
}
```

Heat map function
```{r}
myHeatmap <- function(genes, title) {
  # this function takes in the read count matrix (normalized and transformed) and a list of gene IDs
  # and plots the values stratified by genotype and timepoint
  names = names(genes)
  if(any(!genes %in% rownames(se))){
    print("The following genes are not included in the experiment. Check to make sure that the gene names are correct.")
    print(setdiff(genes, rownames(se)))
    names = names[genes %in% rownames(se)]
    genes = intersect(genes, rownames(se))
  }
  
  # construct tibble for plotting
  exn <- t(assay(se[genes,])) %>% 
    as_tibble(rownames = NA) %>% 
    rownames_to_column(var = "sample") # subset the expression matrix
  
  # append gene names and sample time info
  tb <- colData(se) %>% 
    as_tibble(rownames = NA) %>% 
    rownames_to_column(var = "sample") %>% 
    select(sample, Genotype, Timepoint) %>%
    left_join(exn, by = "sample") %>% 
    pivot_longer(cols = starts_with("CAGL"), names_to = "gene", values_to = "log2 cpm") %>% 
    left_join(tibble(gene = genes, name = names), by = "gene")
    
  # calculate the mean for each gene at each time point
  dat <- group_by(tb, Timepoint, gene, name) %>% 
    summarize(log2 = mean(`log2 cpm`), .groups = "drop") %>% 
    group_by(gene) %>% 
    mutate(log2FC = log2 - log2[Timepoint == "pre"]) %>% 
    filter(Timepoint != "pre")
  
  # construct matrix for plotting
  mat <- dat %>% ungroup() %>% 
    mutate(rowname = paste(gene, name, sep = "_"),
           time = gsub("m$", "", Timepoint)) %>% 
    select(-gene, -name, Timepoint) %>% 
    pivot_wider(id_cols = rowname, names_from = time, values_from = log2FC) %>% 
    column_to_rownames("rowname")
    
  # plotting with pheatmap
  file = glue::glue("../output/heatmap/", gsub("-", "", Sys.Date()), "-",
                    gsub(" ", "-", tolower(title)), ".png")
  pheatmap(mat, 
           color = colorRampPalette(c("blue", "white", "red"), space="rgb")(64),
           clustering_method = "ward.D",
           breaks = seq(from = -2.1, to = 2.1, length.out = 65),
           border_color = NA, cluster_cols = FALSE, treeheight_row = 0,
           cellwidth = 20, cellheight = 20, main = title,
           filename = file)
}
```

```{r, fig.width=5, fig.height=6}
for (i in seq_along(categories)){
  myHeatmap(categories[[i]], str_to_upper(group[i]))
}
```

> **Adhesin genes**
>  
> - only 14 of the 24 were present in my dataset due to an older genome annotation. Among them, six showed induction. Caveat: missing many, and there could be mapping errors due to sequence similarity.
>
> **Cell cycle arrest, DNA repair, DNA damage checkpoint**
>
> - No clear pattern
>
> **Oxidative stress response**
>
> - Of the ~20 genes they listed, half are strongly induced. These include CTA1 and Msn4. The other half are not the same ones we looked at (based on Hasan 2002 and Lee 1999). Overall, there is a strong signature of OSR induction.
>
> **Autophagy**
>
> - More than 2/3 of the autophagy genes in the ref list are strongly induced. Very interesting.
>
> **TCA cycle**
>
> - More than 80% of the TCA cycle genes were induced. Recall that TCA cycle (also called Citric Acid Cycle or Krebs cycle) is closely linked to the oxidative phosphorylation pathway and together are responsible for converting nutrients such as glucose to energy using respiration. C. glabrata like S. cerevisiae, prefers to ferment. However, glucose is highly limited inside the phagosome and it would make sense for the yeast to induce TCA cycle in preparation for alternative carbon usage. Why -Pi induces TCA, however, is something less clear, since phosphate starvation would actually impose challenges for the upper glycolysis, which is an ATP-consuming process. Nonetheless, the induction of TCA cycle is one of the three most prominent similarities between -Pi and macrophage
>
> **Amino acid biosynthesis**
>
> - 1/3 of the genes induced, weak/unclear pattern.
>
> **Iron homeostasis**
>
> - 1/3 of the genes strongly induced, 1/4 strongly repressed, others weak.

### Compare with _S. cerevisiae_ -Pi at 1hr

To compare the induction of genes associated with phagocytosis under phosphate starvation in _C. glabrata_ and _S. cerevisiae_, I downloaded Xu's 2011 data, which was measured using two-color microarray at 1hr post starvation. To make the two species comparable, I will extract the 1hr time point data from my time course data set for _C. glabrata_.

First, we need to assemble the dataset. The idea is to calculate the log2FC between 1hr and pre-stress time points from the _C. glabrata_ dataset. Later we will merge it with the _S. cerevisiae_ data. The final data should include `Species | Name | ID | log2FC`, in which `Name` is always from _S. cerevisiae_ and `ID` from the respective species.
```{r}
tmp <- assay(se[, se$Timepoint %in% c("pre", "60m")])
colnames(tmp) <- c("pre.1","pre.2","60m.1","60m.2","60m.3")
cgla.1hr <- as_tibble(tmp, rownames = "CglaID") %>% 
  mutate(
    base = (pre.1 + pre.2)/2,
    m60.1 = `60m.1` - base,
    m60.2 = `60m.2` - base,
    m60.3 = `60m.3` - base
  ) %>% 
  select(CglaID, starts_with("m60")) %>% 
  pivot_longer(m60.1:m60.3, names_to = NULL, values_to = "log2FC")
```

Note that this calculation "hides" the variation in the pre-stress sample by using only the mean level of expression. As a result, the three replicate values are actually an underestimate of the amount of variation in _C. glabrata_. By contrast, the _S. cerevisiae_ dataset were measured using two-color microarrays, which are naturally paired -- it always measures the ratio between the stressed and unstressed samples. That being said, I don't think the magnitude of the underestimation is going to be large, given the tight correlations between the biological replicates. And the overall conclusion from this figure is not going to change -- a group of OSR genes are strongly induced in _C. glabrata_ but not in _S. cerevisiae_

With the Cgla data calculated, we can write a function that takes a list of _S. cerevisiae_ gene names as input and plot the logFC in both species.
```{r}
myCompare <- function(genes = "CAGL0B02475g"){
  # ------
  # this function is used to extract and assemble the expression data from the two species
  # it returns a tibble that can be used for plotting
  # ------
  
  # C. glabrata gene names are available as
  cglaID <- cgla.1hr$CglaID
  
  # check gene names against the C. glabrata gene set
  if(all(!genes %in% cglaID)){
    stop("Check ORF names. None are present in the annotation file.")
  }else if(any(!genes %in% cglaID)){
    print("The following C. glabrata gene IDs do not match anything in the dataset.")
    print(setdiff(genes, cglaID))
  }else if(any(!genes %in% re.scToCg$CglaID)){
    # additionally check if the genes have a S. cerevisiae ortholog
    print("The following C. glabrata genes do not have a S. cerevisiae ortholog.")
    print(setdiff(genes, re.scToCg$CglaID))
  }
  
  # construct a gene set tibble that includes orthologous gene names in both species
  geneSet <- filter(re.scToCg, CglaID %in% genes)
  
  # extract Scer data
  probes <- rowData(gse1) %>% as_tibble() %>% filter(ORF %in% geneSet$ScerID) %>% select(ID, ORF)
  scerDat <- assay(gse1)[probes$ID,,drop=FALSE] %>% 
    as_tibble(rownames = "ID") %>% 
    left_join(probes, by = "ID") %>% 
    pivot_longer(cols = starts_with("GSM"), names_to = NULL, values_to = "log2FC") %>% 
    select(ScerID = ORF, log2FC) %>% 
    left_join(select(geneSet, ScerID, ScerName, CglaName), by = "ScerID") %>% 
    select(ScerName, CglaName, ID = ScerID, log2FC)
  # extract Cgla data
  cglaDat <- cgla.1hr %>% 
    filter(CglaID %in% na.omit(geneSet$CglaID)) %>% 
    left_join(select(geneSet, CglaID, ScerName, CglaName), by = "CglaID") %>% 
    select(ScerName, CglaName, ID = CglaID, log2FC)
  Dat <- bind_rows(`S. cerevisiae` = scerDat, `C. glabrata` = cglaDat, .id = "species") %>%
    # sort the gene list by Cgla induction mean
    mutate(ScerName = forcats::fct_reorder(ScerName, log2FC, mean, .desc = TRUE))
  return(Dat)
}
```

```{r}
myPlotExn <- function(dat, col = c("S. cerevisiae" = "gray30", "C. glabrata" = "orange2"),
                      #fill = c("S. cerevisiae" = "#61328C", "C. glabrata" = "#5B9A3D"),
                      fill = c("S. cerevisiae" = "gray60", "C. glabrata" = "#FBB4AE"),
                      ylim, font_size = 16, legend = TRUE, grid = FALSE)
{
  ## -------
  ## this function creates a reproducible plot allowing flexibility in color and legend
  geoms <- list(
    geom_bar(aes(fill = species), stat = "summary", fun = "mean",width = 0.8,
             color = "grey20", linewidth = 1, position = position_dodge(0.9)),
    geom_point(aes(color = species), position = position_dodge(0.9), size = 1),
    geom_hline(yintercept = 0, linetype = 1, color = "gray20"),
    geom_hline(yintercept = 1, linetype = 2, color = "gray50"),
    scale_color_manual(values = col, guide = NULL),
    scale_y_continuous(limits = ylim),
    scale_fill_manual("Species", values = fill),
    labs(x = expression(paste(italic("S. cerevisiae"), " gene names")),
         y = "log2 Fold Change"),
    theme_bw(base_size = font_size),
    theme(legend.title = element_blank(),
          legend.text = element_text(face = 3), 
          legend.position = c(0.8, 0.8),
          legend.background = element_blank(),
          axis.text.y = element_text(size = rel(1)),
          axis.text.x = element_text(angle = 45, hjust = 1, size = rel(0.9)),
          axis.title = element_text(size = rel(1)),
          axis.title.x = element_blank(),
          plot.title = element_text(size = rel(1.5))),
    if(!legend) theme(legend.position = "none"),
    if(!grid) background_grid(major = "none", minor = "none")
  )
  p <- ggplot(dat, aes(x = ScerName, y = log2FC, group = species)) + geoms
  return(p)
}
```

Decide on the list of gene categories to plot
```{r}
group <- c(autophagy = "autophagy", tca = "TCA cycle", amino = "amino acid biosynthesis", 
           iron = "iron homeostasis")
geneList <- lapply(group, function(x){
  cgla.phage %>% filter(category == x) %>% pull(geneID)
})
names(geneList) <- names(group)
```

Get the dataset for plotting
```{r}
dat <- lapply(geneList, myCompare)
```

Make individual plots
```{r}
par.fill = c("S. cerevisiae" = "white", "C. glabrata" = "gray40")
par.col = c("S. cerevisiae" = "#6aa84f", "C. glabrata" = "orange3")
p.obj <- list(
  myPlotExn(dat$autophagy, legend = TRUE, fill = par.fill, col = par.col, ylim = c(-3, 4.5)),
  myPlotExn(dat$tca, legend = FALSE, fill = par.fill, col = par.col, ylim = c(-2.5, 3)),
  myPlotExn(dat$iron, legend = FALSE, fill = par.fill, col = par.col, ylim = c(-2.5, 3)),
  myPlotExn(dat$amino, legend = FALSE, fill = par.fill, col = par.col, ylim = c(-2, 2))
)
names(p.obj) <- c("autophagy", "tca", "iron", "amino")
row1 = plot_grid(p.obj$autophagy, p.obj$amino + theme(axis.title.y = element_blank()), 
                 nrow = 1, align = "hv", rel_widths = c(1.8, 1))
row2 = plot_grid(p.obj$tca, p.obj$iron + theme(axis.title.y = element_blank()),
                 nrow = 1, align = "hv", rel_widths = c(1, 1))
plot_grid(row1, row2, nrow = 2)
ggsave("../output/20230502-phage-gene-in-noPi-compare.png", width = 8, height = 6)
```

### Statistical tests for DGE
The bar plots above clearly show the differences between the two species in terms of OSR gene induction under phosphate starvation in the two species. It is likely that reviewers would ask for statistical test results to support the trend, i.e., whether the gene inductions were significant and whether the differences between species were significant. Below, we perform simple t-tests to provide such support. Note the caveat due to the subtraction of the mean of the pre-stress sample from the 1hr samples in _C. glabrata_, which is expected to slightly decrease the p-values, as the variance would be underestimated. I don't expect this to have a meaningful impact on the ranking of the genes, given the clear difference in the magnitude of induction between the two species.

First, we need to prepare the data. We would like to get a single dataset with all replicates for each gene in the two species.
```{r}
dge.dat <- bind_rows(dat, .id = "group")
```

T-test for each gene against the null hypothesis of no differential expression
```{r}
dge.test <- dge.dat %>% group_by(species, ScerName) %>% 
  nest() %>% 
  mutate(test = map(data, ~ t.test(.x$log2FC, alternative = "greater")),
         tidied = map(test, tidy)) %>% 
  unnest(tidied) %>% 
  select(species, ScerName, lfc = estimate, t.stat = statistic, p.value)
dge.test$p.adj <- p.adjust(dge.test$p.value, method = "BH")
dge.test %>% filter(p.adj < 0.05, lfc >= 1) %>% arrange(desc(lfc), p.value)
```



### Plot _C. glabrata_ infection time course for PHO genes
Here we will directly use the normalized PolII ChIP-seq RPKM data
```{r}
geneOrder = c("CAGL0B02475g", "CAGL0C02321g", "CAGL0M12705g", "CAGL0K07546g", "CAGL0K07524g",
              "CAGL0M12430g", "CAGL0M11660g", "CAGL0F02387g", "CAGL0G06952g", "CAGL0L05456g")
p <- chip.phage %>% 
  pivot_longer(cols = ends_with("h"), names_to = "time", values_to = "x") %>% 
  group_by(geneID) %>% 
  mutate(norm = x / mean(x)) %>% 
  ungroup() %>% 
  mutate(geneID = fct_relevel(geneID, geneOrder),
         time = gsub("h", "", time) %>% as.numeric(), 
         gene = paste(geneID, geneName, sep = "_")) %>% 
  ggplot(aes(x = time, y = norm)) +
  geom_line() + geom_point() +
  scale_y_continuous(breaks = c(0, 1, 2), limits = c(-.1, 2.3)) +
  xlab("Time (hrs)") + ylab("ChIP RPKM (mean normalized)") +
  theme_minimal(base_size = 14) + 
  panel_border(color = "black", size = 0.5) +
  background_grid(minor = "y", major = "y")

p + facet_wrap(~geneID, nrow = 3)

ggsave("../output/20230502-cgla-PHO-genes-in-infection.png", 
       p + facet_wrap(~geneID, ncol = 5), width = 7, height = 5)
```

